posts/onboarding.md

```diff
@@ -0,0 +1,525 @@
+            ...
+        }
+```
+
+
+Even though this example is a fairly realistic starting point, we strongly encourage you to study existing dialect implementations to understand how their various components can be modified.
+
+
+## Schema
+Previous sections described the SQLGlot components used for transpilation. This section and the next describe components needed to implement other SQLGlot functionality like column-level lineage.
+
+A Schema represents the structure of a database schema, including the tables/views it contains and their column names and data types.
+
+The file `schema.py` defines the abstract classes `Schema` and `AbstractMappingSchema`; the implementing class is `MappingSchema`.
+
+A `MappingSchema` object is defined with a nested dictionary, as in this example:
+
+```Python
+schema = MappingSchema({"t": {"x": "int", "y": "datetime"}})
+```
+
+The keys of the top-level dictionary are table names (`t`), the keys of `t`’s dictionary are its column names (`x` and `y`), and the values are the column data types (`int` and `datetime`).
+
+A schema provides information required by other SQLGlot modules (most importantly, the optimizer and column level lineage).
+
+Schema information is used to enrich ASTs with actions like replacing stars (`SELECT *`) with the names of the columns being selected from the upstream tables.
+
+## Optimizer
+
+The optimizer module in SQLGlot (`optimizer.py`) is responsible for producing canonicalized and efficient SQL queries by applying a series of optimization rules.
+
+Optimizations include simplifying expressions, removing redundant operations, and rewriting queries for better performance.
+
+The optimizer operates on the abstract syntax tree (AST) returned by the parser, transforming it into a more compact form while preserving the semantics of the original query.
+
+> [!NOTE]
+> The optimizer performs only logical optimization; The underlying engine is almost always better at optimizing for performance.
+
+
+### Optimization rules
+The optimizer essentially applies [a list of rules](https://sqlglot.com/sqlglot/optimizer.html) in a sequential manner, where each rule takes in an AST and produces a canonicalized and optimized version of it.
+
+> [!WARNING]
+> The rule order is important, as some rules depend on others to work properly. We discourage manually applying individual rules, which may result in erroneous or unpredictable behavior.
+
+The first, foundational optimization task is to _standardize_ an AST, which is required by other optimization steps. These rules implement canonicalization:
+
+#### Qualify
+The most important rule in the optimizer is `qualify`, which is responsible for rewriting the AST such that all tables and columns are _normalized_ and _qualified_.
+
+##### Identifier normalization
+The normalization step (`normalize_identifiers.py`) transforms all unquoted identifiers to lower or upper case (depending on the dialect), essentially making them case-insensitive.
```

Comment:
```suggestion
The normalization step (`normalize_identifiers.py`) transforms some identifiers, by converting them into lower or upper case, ensuring the semantics are preserved in each case (e.g. by respecting case-sensitivity).

This transformation reflects how identifiers would be resolved by the engine corresponding to each SQL dialect, and plays a very important role in the standardization of the AST.
```

Reply:
The explanations of these optimizer steps originate mostly from the docstrings (this one is from `normalize_identifiers.py` in particular) as I didn't want to stray off; If they're outdated or require new details to be up to date, we can also update them too (?)

Reply:
Yep, they're probably inaccurate / outdated and we should fix them.